### PR TITLE
Changed behaviour of Bresenham allgorithm

### PR DESCRIPTION
**Pull request by ravifrancesco**
Fixed a bug that caused the Bresenham algorithm to create empty lines of points in some cases.

Now the line of sight generates the correct line every for every POIPoint.

**TODO**
Fix other bugs that cause the line of sight to not work properly

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/peakar/points/LineOfSight.java [hunk 1]
```diff
@@ -152,11 +153,13 @@ private boolean isVisible(POIPoint poiPoint, Pair<Integer, Integer> userIndexes,
...
             if (slope_error_new >= 0) {
-                y++;
-                slope_error_new -= 2 * (x2 - x1);
+                y = y2 > y1 ? y + 1 : y - 1;
+                slope_error_new -= 2 * Math.abs(x2 - x1);
             }
         }
 
+
+
```
Comment (giommok) [inline app/src/main/java/ch/epfl/sdp/peakar/points/LineOfSight.java:162]: Empty lines
```suggestion
```

File: app/src/main/java/ch/epfl/sdp/peakar/points/LineOfSight.java [hunk 2]
```diff
@@ -152,11 +153,13 @@ private boolean isVisible(POIPoint poiPoint, Pair<Integer, Integer> userIndexes,
             // Slope error reached limit, time to
             // increment y and update slope error.
```
Comment (giommok) [inline app/src/main/java/ch/epfl/sdp/peakar/points/LineOfSight.java:154]: I think the comment was describing the old algorithm.
```suggestion
            // increment or decrement y and update slope error.
```

**Review by giommok — CHANGES_REQUESTED** [review]
Good job at finding that bug!
I left two comments.

**Review by bgabriel998 — APPROVED** [review]
Not much to say 👍

**Review by giommok — APPROVED** [review]

**Pull request merged**